tests/unittests/test_datasource/test_azure.py

```diff
@@ -384,6 +411,41 @@ def test_get_metadata_does_not_dhcp_if_network_is_up(
             "Crawl of Azure Instance Metadata Service (IMDS) took",  # log_time
             self.logs.getvalue())
 
+    @mock.patch(MOCKPATH + 'readurl')
```

Comment:
To use the `assert_*` methods, we require autospeccing:

```suggestion
    @mock.patch(MOCKPATH + 'readurl', autospec=True)
```